diff --git a/apps/lib/apps.c b/apps/lib/apps.c
index 50e83b50c4..752c32cf90 100644
--- a/apps/lib/apps.c
+++ b/apps/lib/apps.c
@@ -2889,7 +2889,7 @@ double app_tminterval(int stop, int usertime)
     return ret;
 }
 
-#elif defined(_SC_CLK_TCK)      /* by means of unistd.h */
+#elif defined(_SC_CLK_TCK) && !defined(__EMSCRIPTEN__)     /* by means of unistd.h */
 # include <sys/times.h>
 
 double app_tminterval(int stop, int usertime)
diff --git a/apps/speed.c b/apps/speed.c
index 059183ddc7..f3e2fa55b3 100644
--- a/apps/speed.c
+++ b/apps/speed.c
@@ -118,7 +118,7 @@ typedef struct openssl_speed_sec_st {
 static volatile int run = 0;
 
 static int mr = 0;  /* machine-readeable output format to merge fork results */
-static int usertime = 1;
+static int usertime = 0;
 
 static double Time_F(int s);
 static void print_message(const char *s, int length, int tm);
@@ -512,7 +512,12 @@ static size_t sigs_algs_len = 0;
 static char *sigs_algname[MAX_SIG_NUM] = { NULL };
 static double sigs_results[MAX_SIG_NUM][3];  /* keygen, sign, verify */
 
+#ifdef __EMSCRIPTEN__
+#include "sched.h"
+#define COND(unused_cond) (sched_yield(), run && count < (testmode ? 1 : INT_MAX))
+#else
 #define COND(unused_cond) (run && count < (testmode ? 1 : INT_MAX))
+#endif
 #define COUNT(d) (count)
 
 #define TAG_LEN 16 /* 16 bytes tag length works for all AEAD modes */
